sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6524,14 +6524,21 @@ pi_result piEnqueueEventsWaitWithBarrier(pi_queue Queue,
       for (ze_command_queue_handle_t ZeQueue : QueueGroup.ZeQueues) {
         pi_command_list_ptr_t CmdList;
         if (auto Res = Queue->Context->getAvailableCommandList(
-                Queue, CmdList, UseCopyEngine, OkToBatch, &ZeQueue))
+                Queue, CmdList, UseCopyEngine, OkToBatch,
+                ZeQueue ? &ZeQueue : nullptr))
```

Comment:
This will allocate multiple command-lists for multiple ZeQueues that are nullptr 
I think for queues where no previous command-lists were submitted we just don't need any barriers inserted for them

Reply:
Yes, now we do nothing for unused queues.